### PR TITLE
[NP-3512] Use Design System components

### PR DESCRIPTION
**Pull request by leeky**

### CODE REVIEW

File: lib/citizens_advice_form_builder/form_builder.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      label ||= object.class.human_attribute_name(attribute)
+
+      optional = if required
+                   false
+                 else
+                   true
+                 end
+
+      component = CitizensAdviceComponents::TextInput.new(
+        name: id_for(attribute),
```
Comment (marianayovcheva) [inline lib/citizens_advice_form_builder/form_builder.rb:24]: The file's logic makes sense and I think it's good to go as it is. 
I did a quick spike in design system to see if there is something we can do to make it easier to assign an id https://github.com/citizensadvice/design-system/compare/main...spike/input-id. The option in the spike would be non-breaking and if we go for it we can just define the `id: id_for` here and remove the `additional_attributes` line. Let me know what you think and based on that I can then turn it into a proper PR

**Comment by marianayovcheva** [comment]
Looking great! 🎉 I've also tried it out locally with the and it worked for me. Let me know once it's undrafted and I'll approve it.

File: lib/citizens_advice_form_builder/form_builder.rb (hunk 1 above)
Comment (leeky) [reply lib/citizens_advice_form_builder/form_builder.rb:24]: Adding `id` as a new parameter to the components makes a lot of sense to me!

Comment (marianayovcheva) [reply lib/citizens_advice_form_builder/form_builder.rb:24]: Great! I can implement it once I'm done with my current ticket but let me know if it's blocking you and I can look into adding it earlier.

Comment (leeky) [reply lib/citizens_advice_form_builder/form_builder.rb:24]: I can work with what I have for now, so it's not blocking me - it's more that making being able to override the `id` parameter on the components means I will no longer have to fudge it to work!

Thanks again :)

**Review by marianayovcheva — APPROVED** [review]

**Pull request merged**